lib/scala/parser/src/main/scala/org/enso/parser/Parser.scala

```diff
@@ -0,0 +1,24 @@
+package org.enso.parser
+
+import org.enso.ast.Ast
+
+import scala.annotation.nowarn
+
+
+/** This is the Enso language parser.
+  *
+  * It is a wrapper of parser written in Rust that uses JNI to efficiently
+  * construct scala AST directly without any serialization overhead.
+  */
+class Parser private () {
+  /**  Parses a single sourcef file. */
+  @native @nowarn def parse(input: String): Ast.AnyAst
+}
+
+object Parser {
+  /** Constructs a new parser */
+  def apply(): Parser = {
+    System.loadLibrary("parser")
```

Comment:
Isn't it enough to call `loadLibrary` only on the first call?

Reply:
It should be placed in `static { .. }`, but scala doesn't have that. I have found a workaround though.